grisp/default/common/build/nifs/grisp_rtems_nif.c

```diff
@@ -95,10 +98,32 @@ static ERL_NIF_TERM clock_set(ErlNifEnv *env, int argc,
   }
 }
 
+static ERL_NIF_TERM unmount_nif(ErlNifEnv *env, int argc,
+                                const ERL_NIF_TERM argv[]) {
+  char path[0xFF];
+  int len = 0;
+  int status;
+
+  if ((len = enif_get_string(env, argv[0], path, sizeof(path),
```

Comment:
It's better to use iolists, like so:
```
ErlNifBinary path;
...
if (!enif_inspect_iolist_as_binary(env, argv[0], &path)) {
  return RAISE_TERM(am_invalid_path, argv[0]);
}
...
unmount((char *)path.data);
```
(Example: https://github.com/grisp/grisp/blob/master/grisp/default/common/build/nifs/grisp_ni2c_nif.c#L93-L95)

This way, you can get rid of the guard on the Erlang function too (since the NIF will throw the specific error.

Reply:
It's not just a matter of taste, I think. It means you can be more flexible (and performant) with the creation of the string input:

```erl
grisp_rtems:unmount([Root, <<"/foo">>, "/bar"]).
```

Making an iolist null terminated in Erlang is fast and simple: `[Path, 0]`.

Reply:
Why do you want to do stuff that's only relevant for C in Erlang? I absolutely get that for performance reasons, but everything else regarding flexibility or whatever can be done in Erlang. And I'm not optimizing here for `unmounts/sec` ;).

No hard feelings though, if that's some sort of inofficial string handling we have then I change it.

Reply:
That's my point: it is _not_ only relevant for C. Iolists are already used with sockets, file APIs etc. in Erlang. It's much more convenient and efficient to use them here too.

It's the same amount of code. It takes literally 5 minutes to change, and we already use it elsewhere so the code will be more homogenous. I frankly don't see a reason _not_ to use it (unless you really can't spare 5 minutes).

Reply:
And frankly, ignore the efficiency argument for now. For me it is mostly about convenience, so I can avoid sprinkling `filename:join()` and `++` in the Erlang code.